compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -173,7 +173,23 @@ pub fn opts(os: &'static str, arch: Arch) -> TargetOptions {
     }
 }
 
-fn deployment_target(var_name: &str) -> Option<(u32, u32)> {
+pub fn deployment_target(target: &Target) -> String {
+    let (major, minor) = match &*target.os {
+        "macos" => {
+            // This does not need to be specific. It just needs to handle x86 vs M1.
+            let arch = if target.arch == "x86" || target.arch == "x86_64" { X86_64 } else { Arm64 };
+            macos_deployment_target(arch)
+        }
+        "ios" => ios_deployment_target(),
+        "watchos" => watchos_deployment_target(),
+        "tvos" => tvos_deployment_target(),
+        _ => unreachable!("unknown Apple target OS"),
```

Comment:
This arm is reachable with custom target specs.
I suggest returning an `Option` here, and relying on it in the driver instead of using `sess.target.is_like_osx`.

Reply:
Ah, thanks, and good to know. Moved up to `rustc_driver`. For the future, why is doing it there better? Do custom target specs go through a different codepath that wouldn't panic?